src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -114,4 +205,343 @@
+            }
+            else if (span == SpanType.WEEK || span == SpanType.MONTH) {
+                index = date.getDayOfMonth();
+            }
+            else if (span == SpanType.QUARTER) {
+                index = getWeekOfDate(date);
+            }
+            else {
+                index = date.getMonth();
+            }
+            String username = listElement.get("username").toString();
+            List<String> usersInSameSlot = users.get(index);
+            // if this index is not yet existing in users
+            if (usersInSameSlot == null) {
+                usersInSameSlot = new ArrayList<>();
+                usersInSameSlot.add(username);
+                users.put(index, usersInSameSlot);
+            }   // if the value of the map for this index does not contain this username
+            else if (!usersInSameSlot.contains(username)) {
+                usersInSameSlot.add(username);
+            }
+        }
+        return fillMapList(users, span, startDate);
+    }
+
+    /**
+     * Helper class for the ConvertMapList method, which takes the users in the same timeslot as well as some attributed needed
+     * for calculation to convert these into a Map List which is then returned
+     *
+     * @param users the List of Maps each containing the date and a list of users
+     * @param span the spanType for which we created the users List
+     * @param startDate the startDate which we need for mapping into timeslots
+     * @return A List<Map<String, Object>> with no duplicated user per timeslot
+     */
+    private List<Map<String, Object>> fillMapList(Map<Object, List<String>> users, SpanType span, ZonedDateTime startDate) {
+        List<Map<String, Object>> returnList = new ArrayList<>();
+        users.forEach((k, v) -> {
+            Object start;
+            if (span == SpanType.DAY) {
+                start = startDate.withHour((Integer) k);
+            }
+            else if (span == SpanType.WEEK || span == SpanType.MONTH) {
+                start = startDate.withDayOfMonth((Integer) k);
+            }
+            else if (span == SpanType.QUARTER) {
+                int year = (Integer) k < getWeekOfDate(startDate) ? startDate.getYear() + 1 : startDate.getYear();
+                ZonedDateTime firstDateOfYear = ZonedDateTime.of(year, 1, 1, 0, 0, 0, 0, startDate.getZone());
+                start = getWeekOfDate(firstDateOfYear) == 1 ? firstDateOfYear.plusWeeks(((Integer) k) - 1) : firstDateOfYear.plusWeeks((Integer) k);
+            }
+            else {
+                start = startDate.withMonth(getMonthIndex((Month) k));
+            }
+            Map<String, Object> listElement = new HashMap<>();
+            listElement.put("day", start);
+            listElement.put("amount", (long) v.size());
+            returnList.add(listElement);
+        });
+        return returnList;
+    }
+
+    private Integer getMonthIndex(Month month) {
+        return switch (month) {
```

Comment:
please delete this method and use the method `getValue()` in month. It does exactly what you want

Reply:
Thanks! Did not know this method existed